src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
         )
 
 
+@dataclass
```

Comment:
I love it! I wonder though, if you could make it immutable by changing some small things in the connection functionality. I'll add some thoughts below.